### DISCUSSION TITLE
List of error or status codes

### DISCUSSION THREAD

**Discussion by ausminternet** [pocketbase/pocketbase#3922] [Q&A]
Hey there,

I already looked up the documentation, but I can’t find a list of the possible error codes for the JavaScript api calls. From the source code I have seen, it seems that it’s always a ClientResponseError with no special typing. 

Can someone point me to the right direction? A list of all available errors and/or status codes for the various api calls?

thx Jan

**Comment by ganigeorgiev** [comment]
Another user also requested something similar in https://github.com/pocketbase/pocketbase/issues/3798, but unfortunately there is no such list at the moment and the validation errors will depend on the specific collection fields options.

Once we stabilize the APIs I'll make sure to document them, but for now you'll have to rely on a trial and error approach.

In general all API failures have a status code >= 400 and all share the same json response structure:

```js
{
    // response status
    "code": 400,

    // general error message
    "message": "Something went wrong while processing your request.",
    
    // additional error data containing fields validation errors
    "data": {
        "title": {
            "code": "validation_length_invalid",
            "message": "The length must be exactly 255."
        }
    }
}
```

To some extend one possible workaround to extract the current validation error codes could be to try to search for [`validation.NewError(` in the GitHub repo](https://github.com/search?q=repo%3Apocketbase%2Fpocketbase++validation.NewError%28&type=code).

**Reply by ganigeorgiev** [reply]
_Side-note: if you are asking for the available fields of the JS SDK `ClientResponseError` you could check https://github.com/pocketbase/js-sdk#error-handling (`ClientResponseError.response` returns the failed API JSON response)._
